config/core/roles/controller-clusterroles.yaml

```diff
@@ -125,3 +125,11 @@ rules:
       - "get"
       - "list"
       - "watch"
+
+  # The source CRD controller needs to also patch CustomResourceDefinitions finalizers
+  - apiGroups:
+      - "apiextensions.k8s.io"
+    resources:
+      - "customresourcedefinitions/finalizers"
```

Comment:
I do not think we want to set or adjust CRD finalizers.

Reply:
this is to be able to delete dynamically created controllers whenever a source CRD gets deleted... do you think we don't need this at all?

Reply:
I think we should handle non-recoverable errors and delete the controllers if they happen rather than block CRD deletion. The registry is not that important and it would suck to delete Knative and then have to edit the CRDs I used as a source that are not related to knative.

Reply:
Valid point, agree. Let me remove this stuff. Thanks Scott!